lib/contract/validateDPContractFactory.js

```diff
@@ -20,10 +24,75 @@ module.exports = function validateDPContractFactory(validator) {
     // TODO: Use validateSchema
     //  https://github.com/epoberezkin/ajv#validateschemaobject-schema---boolean
 
-    return validator.validate(
+    const result = validator.validate(
       JsonSchemaValidator.SCHEMAS.META.DP_CONTRACT,
       rawDPContract,
     );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // Validate indices
+    Object.entries(rawDPContract.dpObjectsDefinition).filter(([, dpObjectDefinition]) => (
+      Object.prototype.hasOwnProperty.call(dpObjectDefinition, 'indices')
+    ))
+      .forEach(([dpObjectType, dpObjectDefinition]) => {
+        const indicesFingerprints = [];
+
+        dpObjectDefinition.indices.forEach((indexDefinition) => {
+          const indexPropertyNames = Object.keys(indexDefinition.properties);
+
+          const indicesFingerprint = JSON.stringify(indexDefinition.properties);
```

Comment:
Do you think that such a fingerprint is enough? What will happen if we define properties in a different order, but the index would be the same?